### DISCUSSION TITLE
Terminating an app once app.Run() is called

### DISCUSSION THREAD

**Discussion by astdb** [progrium/darwinkit#89] [Q&A]
How do I go about stopping an app programmatically once its Run() method is called?

For example,   

```
func main() {
	app := cocoa.NSApp_WithDidLaunch(func(notification objc.Object) {
		config := webkit.WKWebViewConfiguration_New()
		.......
	})
	app.SetActivationPolicy(cocoa.NSApplicationActivationPolicyRegular)
	app.ActivateIgnoringOtherApps(true)
	app.Run()

       app.Terminate()
}

```
Calling app.Terminate() won't have an effect after the Run() call. What's the best way to achieve it?

Some context on why I need to do this. I have a legacy Go-written app which has a menubar icon. It uses defaults AppleInterfaceSyle to determine menubar color (dark or light) - which doesn't work post- Big Sur operating systems. I'm using Macdriver to create a button and check for EffectiveAppearance to get an indication of if the menubar is light or dark. As far as I know this is the only method to test this when the menubar is auto-rendered dark/light based on background wallpaper color. Ref: https://stackoverflow.com/questions/62685948/macos-big-sur-detect-dark-menu-bar-system-tray

Once I run the app and detect the color I'd like to quit the app and continue on with the rest of the main.go

**Comment by progrium** [comment]
Did you try:

```golang
func main() {
	app := cocoa.NSApp_WithDidLaunch(func(notification objc.Object) {
		config := webkit.WKWebViewConfiguration_New()
		.......
		app.Terminate()
	})
	app.SetActivationPolicy(cocoa.NSApplicationActivationPolicyRegular)
	app.ActivateIgnoringOtherApps(true)
	app.Run()
}
```

**Comment by progrium** [comment]
I don't think I noticed you wanted to continue the program after `Run()`. I don't believe this is a common idiom for Objective-C using Appkit, so I'm not sure there is an API to do this. However, you could try starting a goroutine that blocks until you signal it. Just keep in mind if you use frameworks from it, you'll need an autorelease pool and interacting with existing objects on the main thread will need to use dispatch so the operation happens in the main thread.
